core-tests/src/test/java/org/mule/runtime/core/mule/enricher/MessageEnricherTestCase.java

```diff
@@ -179,11 +159,7 @@ public void propagatesVariables() throws Exception {
     enricher.setEnrichmentMessageProcessor(event -> {
       return Event.builder(event).message(InternalMessage.builder(event.getMessage()).payload("enriched").build()).build();
     });
-    FlowConstruct flowConstruct = MuleTestUtils.getTestFlow(muleContext);
-    Event in = Event.builder(DefaultEventContext.create(flowConstruct, TEST_CONNECTOR))
-        .message(InternalMessage.of(""))
-        .addVariable("flowFoo", "bar")
-        .build();
+    Event in = eventBuilder().message(InternalMessage.of("")).addVariable("flowFoo", "bar").build();
```

Comment:
Just thinking out loud, but another approach to exposing builder and requiring `.build` could be to take a builder function.
`event(payload)`
`event(builder -> builder.addVariable()`